--- a/net/minecraft/world/entity/decoration/EntityItemFrame.java
+++ b/net/minecraft/world/entity/decoration/EntityItemFrame.java
@@ -115,7 +115,7 @@
         return this.createBoundingBox(this.pos, this.getDirection(), false);
     }
 
-    private AxisAlignedBB createBoundingBox(BlockPosition blockposition, EnumDirection enumdirection, boolean flag) {
+    public static AxisAlignedBB createBoundingBox(BlockPosition blockposition, EnumDirection enumdirection, boolean flag) { // CraftBukkit - static
         float f = 0.46875F;
         Vec3D vec3d = Vec3D.atCenterOf(blockposition).relative(enumdirection, -0.46875D);
         float f1 = flag ? 1.0F : 0.75F;
@@ -182,6 +182,11 @@
             if (this.isInvulnerableToBase(damagesource)) {
                 return false;
             } else if (this.shouldDamageDropItem(damagesource)) {
+                // CraftBukkit start - fire EntityDamageEvent
+                if (org.bukkit.craftbukkit.event.CraftEventFactory.handleNonLivingEntityDamageEvent(this, damagesource, f, false) || this.isRemoved()) {
+                    return true;
+                }
+                // CraftBukkit end
                 this.dropItem(worldserver, damagesource.getEntity(), false);
                 this.gameEvent(GameEvent.BLOCK_CHANGE, damagesource.getEntity());
                 this.playSound(this.getRemoveItemSound(), 1.0F, 1.0F);
@@ -294,13 +299,19 @@
     }
 
     public void setItem(ItemStack itemstack, boolean flag) {
+        // CraftBukkit start
+        this.setItem(itemstack, flag, true);
+    }
+
+    public void setItem(ItemStack itemstack, boolean flag, boolean playSound) {
+        // CraftBukkit end
         if (!itemstack.isEmpty()) {
             itemstack = itemstack.copyWithCount(1);
         }
 
         this.onItemChanged(itemstack);
         this.getEntityData().set(EntityItemFrame.DATA_ITEM, itemstack);
-        if (!itemstack.isEmpty()) {
+        if (!itemstack.isEmpty() && playSound) { // CraftBukkit
             this.playSound(this.getAddItemSound(), 1.0F, 1.0F);
         }
 
